scripts-python/case.build

```diff
@@ -0,0 +1,508 @@
+#!/usr/bin/env python
+
+"""
+Script to build a case.
+"""
+
+from standard_script_setup import *
+
+from CIME.utils import expect, run_cmd
+from CIME.case import Case
+from CIME.env_module import EnvModule
+
+import argparse, doctest, shutil, glob
+
+###############################################################################
+def parse_command_line(args, description):
+###############################################################################
+    parser = argparse.ArgumentParser(
+        usage="""\n%s [<casedir>] [--verbose]
+OR
+%s --help
+OR
+%s --test
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# Build case \033[0m
+    > %s
+""" % ((os.path.basename(args[0]), ) * 4),
+
+description=description,
+
+formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    parser.add_argument("caseroot", nargs="?", default=os.getcwd(),
+                        help="Case directory to build")
+
+    parser.add_argument("-t", "--testmode", action="store_true",
+                        help="Unknown TODO")
+
+    args = parser.parse_args(args[1:])
+
+    CIME.utils.handle_standard_logging_options(args)
+
+    return args.caseroot, args.testmode
+
+###############################################################################
+def check_input_data(case):
+###############################################################################
+    run_cmd("./check_input_data --download")
+
+    get_refcase  = case.get_value("GET_REFCASE")
+    run_type     = case.get_value("RUN_TYPE")
+    continue_run = case.get_value("CONTINUE_RUN")
+
+    # JGF - I don't have the foggiest idea what this is for so this is just
```

Comment:
We do not fully populate the inputdata directory on every machine and do not expect every user to download the 3TB+ of data in our inputdata repository.   This code checks for the existence of inputdata  in the local inputdata directory and attempts to download data from the server if it's needed and missing.